File: UnitConverter/app/src/main/java/com/example/unitconverter/MainActivity.java
 package com.example.unitconverter;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText editText;
    TextView textView, textView2, textView4, valueInPounds;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        textView2 = findViewById(R.id.textView2);
        textView4 = findViewById(R.id.textView4);
        valueInPounds = findViewById(R.id.valueOfPounds);
        editText = findViewById(R.id.editTextNumber);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v){
                ConvertFromKiloToPound();
            }
        });
    }

    private void ConvertFromKiloToPound() {
        String kilo = editText.getText().toString();
        double kiloInNum = Double.parseDouble(kilo);
        // converting kilo to pounds formula
        double pounds = kiloInNum * 2.205;
        valueInPounds.setText("" + pounds);
    }
}

File: DiceRoller/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.button)
        rollButton.setOnClickListener() {
            rollDice()
        }
    }
    fun rollDice(){
        // Create new Dice object with 6 sides and roll it
        val dice = Dice(6)
        val diceText = dice.rollDice()
        // Update the screen with the dice roll
        val resultImageView: ImageView = findViewById(R.id.imageView)
        val drawableResource = when (diceText){
            1 -> R.drawable.dice_1
            2 -> R.drawable.dice_2
            3 -> R.drawable.dice_3
            4 -> R.drawable.dice_4
            5 -> R.drawable.dice_5
            else -> R.drawable.dice_6
        }
        resultImageView.setImageResource(drawableResource)
        val ranNumber : EditText = findViewById(R.id.editTextTextPersonName)
        val newRanNumber = ranNumber.text.toString().toInt()
        if(newRanNumber == diceText){
            Toast.makeText(this, "You win!", Toast.LENGTH_SHORT).show()
        }
        else{
            Toast.makeText(this, "Try again!", Toast.LENGTH_SHORT).show()
        }
        //resultImageView.contentDescription = diceText.toString()
    }
}

class Dice(private val numSides: Int) {
    fun rollDice(): Int {
        return (1..numSides).random()
    }

}
